index.js

```diff
@@ -56,128 +60,220 @@ const SCHEMA_SMTP_CONFIG = Joi.object().keys({
+        buildStatus: notificationStatus,
+        buildLink: buildData.buildLink,
+        buildId: buildData.build.id,
+        changedFiles: changedFilesStr,
+        commitSha,
+        commitMessage,
+        commitLink,
+        statusColor: COLOR_MAP[buildData.status]
+    });
 
-                changedFilesStr += tinytim.tim(li, { contents: file });
-            });
-        } else {
-            changedFilesStr = 'There are no changed files.';
-        }
+    const mailOpts = {
+        from: this.config.from,
+        to: Hoek.reach(buildData, 'settings.email.addresses'),
+        subject,
+        text: message,
+        html
+    };
 
-        const ul = '<ul>{{list}}</ul>';
-
-        changedFilesStr = tinytim.tim(ul, { list: changedFilesStr });
-
-        const rootDir = Hoek.reach(buildData, 'pipeline.scmRepo.rootDir', { default: '' });
-        const subject =
-            `${notificationStatus} - Screwdriver ` +
-            `${Hoek.reach(buildData, 'pipeline.scmRepo.name')} ` +
-            `${buildData.jobName} ${rootDir} #${Hoek.reach(buildData, 'build.id')}`;
-        const message = `Build status: ${notificationStatus}\nBuild link:${buildData.buildLink}`;
-        const commitSha = Hoek.reach(buildData, 'build.meta.build.sha').slice(0, 7);
-        const commitMessage = Hoek.reach(buildData, 'build.meta.commit.message');
-        const commitLink = Hoek.reach(buildData, 'build.meta.commit.url');
-        const html = tinytim.renderFile(path.resolve(__dirname, './template/email.html'), {
-            buildStatus: notificationStatus,
-            buildLink: buildData.buildLink,
-            buildId: buildData.build.id,
-            changedFiles: changedFilesStr,
-            commitSha,
-            commitMessage,
-            commitLink,
-            statusColor: COLOR_MAP[buildData.status]
-        });
+    const smtpConfig = {
+        host: this.config.host,
+        port: this.config.port
+    };
 
-        const mailOpts = {
-            from: this.config.from,
-            to: Hoek.reach(buildData, 'settings.email.addresses'),
-            subject,
-            text: message,
-            html
+    if (this.config.username && this.config.password) {
+        smtpConfig.auth = {
+            user: this.config.username,
+            pass: this.config.password
         };
+    }
+
+    emailer(mailOpts, smtpConfig);
+}
+
+/**
+ * Handle email messaging for job status 
+ * @method jobStatus
+ * @param  {Object}         jobData
+ * @param  {String}         jobData.status             Status
+ * @param  {Object}         jobData.pipeline           Pipeline
+ * @param  {String}         jobData.jobName            Job name
+ * @param  {String}         jobData.pipelineLink       Pipeline link
+ * @param  {String}         jobData.message            Message
+ * @param  {Object}         jobData.settings           Notification setting
+ */
+function jobStatus(jobData) {
```

Comment:
non-blocking comment: is it possible to refactor buildStatus/jobStatus methods into one method? they seem pretty similar

Reply:
thing is that they use different info to send the message, only thing we can abstract is prob the part where we send email, but that was only couple lines. If we are to makes them into same functions there would be a lot of if else branch.